Add help for running configure-xcode-for-embedded-development on Ventura

achristensen07 opened the pull request:
#### f0c4152d9e739d0b33e5d7b5482bc46487193332
<pre>
Add help for running configure-xcode-for-embedded-development on Ventura
<a href="https://bugs.webkit.org/show_bug.cgi?id=247032">https://bugs.webkit.org/show_bug.cgi?id=247032</a>

Reviewed by Alexey Proskuryakov and Jonathan Bedard.

* Tools/Scripts/configure-xcode-for-embedded-development:

Canonical link: <a href="https://commits.webkit.org/255994@main">https://commits.webkit.org/255994@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/04b28e16cf7b65a425668c815577d5ca947dbe29

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/94271 "Passed style check") | [  ~~🛠 ios~~](https://ews-build.webkit.org/#/builders/77/builds/3453 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 mac~~](https://ews-build.webkit.org/#/builders/43/builds/26256 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/8/builds/103942 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 🧪 win~~](https://ews-build.webkit.org/#/builders/10/builds/164221 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/98269 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/3489 "Built successfully") | [  ~~🛠 mac-debug~~](https://ews-build.webkit.org/#/builders/71/builds/31666 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/36/builds/86616 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/12/builds/99959 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/99940 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/2501 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 mac-AS-debug~~](https://ews-build.webkit.org/#/builders/61/builds/80672 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/86616 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/84420 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/43/builds/26256 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/36/builds/86616 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/38053 "Built successfully") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/43/builds/26256 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/67/builds/35934 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/43/builds/26256 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/4147 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/39817 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/61/builds/80672 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/41763 "Built successfully") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/43/builds/26256 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
<!--EWS-Status-Bubble-End-->

aproskuryakov commented on Tools/Scripts/configure-xcode-for-embedded-development:
```diff
@@ -227,7 +227,11 @@ def do_copy(source_path, dest_path):
+            shutil.copytree(source_path, dest_path)
+        except:
+            print('Unable to copy needed files. When running this script, please temporarily allow Terminal to update other applications in System Settings > Privacy & Security > App Management.')
+            exit(0)
```
I suggest exit(1), as this is not a success.

aproskuryakov commented on Tools/Scripts/configure-xcode-for-embedded-development:
```diff
@@ -227,7 +227,11 @@ def do_copy(source_path, dest_path):
         if dest_path.exists():
             shutil.rmtree(dest_path)
```
A little surprised that we don't need to change anything here.

JonWBedard replied:
This is mostly for when you're re-running the script...I suppose you could capture the whole thing

JonWBedard commented on Tools/Scripts/configure-xcode-for-embedded-development:
```diff
@@ -227,7 +227,11 @@ def do_copy(source_path, dest_path):
-        shutil.copytree(source_path, dest_path)
+        try:
+            shutil.copytree(source_path, dest_path)
+        except:
```
Not a problem if we only do this for one line, but if we start making this try-except bigger, it's worth being more specific about the exception, this kind of try/except will suppress many errors which would be considered compiler errors in other languages.

Commit:
Add help for running configure-xcode-for-embedded-development on Ventura
https://bugs.webkit.org/show_bug.cgi?id=247032

Reviewed by Alexey Proskuryakov and Jonathan Bedard.

* Tools/Scripts/configure-xcode-for-embedded-development:

Canonical link: https://commits.webkit.org/255994@main
